geosnap/_community.py

```diff
@@ -321,6 +322,25 @@ def cluster_spatial(
         comm.models[model_name] = model
         return comm
 
+    def silplot(self, model_name, year, **kwargs):
+        """ Returns a silhouette plot of the model that is passed to it.
+
+        Parameters
+        ----------
+        model_name : str , required
+                     model to be silhouette plotted
+        year       : dict key, optional
+                     year of model if model has model for each year
+        kwargs     : **kwargs, optional
+                     pass through to plot_silhouette()
+        Returns
+        -------
+        silhouette plot of given model.
+
+        """
+        plot = skplt.metrics.plot_silhouette(self.models[model_name][year].X, self.models[model_name][year].labels, **kwargs)
```

Comment:
we need to add some conditional logic because this will raise if a user wants to plot a model that doesnt have multiple years